src/test_metrics.py

```diff
@@ -68,3 +68,23 @@ def test_worker_status(threaded_exporter, celery_app, hostname):
         )
         == 0.0
     )
+
+
+@pytest.mark.celery()
+def test_worker_heartbeat(threaded_exporter, celery_app, hostname):
+    before_start = time.time()
+    def get_heartbeat_sample():
+        return threaded_exporter.registry.get_sample_value(
+            "celery_worker_last_heartbeat_timestamp", labels={"hostname": hostname}
+        )
+    with start_worker(celery_app, without_heartbeat=False, heartbeat_interval=0.5):
+        time.sleep(2)
+        heartbeat_sample = get_heartbeat_sample()
+        assert heartbeat_sample is not None
```

Comment:
So looking at the debug output it seems like the heartbeat event just never triggers. I've tried sleeping for huge amounts of time at various points in this function, and nothing seems to fix it. If I force the `test_worker_status` function above, which this was largely copied from, to fail and inspect the debug logs I can see that the `worker-heartbeat` event gets sent from correctly in that one. If I move the assertions from this function to that one they pass fine as expected. I don't know why this test fails.

Any ideas?

Reply:
bump on this @danihodovic @adinhodovic, any ideas? I'd like to finalise this so I can start alerting on this new metric without having to run a fork :)